ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerTable.java

```diff
@@ -193,7 +211,6 @@ private void buildBloomFilter() {
   public List<DiscoveryPeer> nearestPeers(final Bytes target, final int limit) {
     final Bytes keccak256 = Hash.keccak256(target);
     return streamAllPeers()
-        .filter(p -> p.getStatus() == PeerDiscoveryStatus.BONDED)
```

Comment:
Why has this method been modified?  And why did we create the very similar `nearestBondedPeers`?  It looks like the functional effect of this is to make the [periodic peer searches](https://github.com/hyperledger/besu/blob/f885c4671333168089a7b86ffedf985fce81954b/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java#L448-L450) run across all peers, not just bonded peers.  Perhaps this is desired, but it should be addressed in a separate PR.  

I suspect the reason this was added is because we're now [directly injecting peers](https://github.com/hyperledger/besu/pull/4178/files#diff-5c9af1f2b004d24055b333ddfc58dc0ee664dcf0649163b140e41d1520df3ce1R356) into the discovery table instead of bonding with them first.  I would suggest reverting these changes, and attempting to bond with the DNS peers rather than directly injecting them.

Reply:
This would be cleaner, I guess. I will do that in a separate PR.